Disable voyager provider in `sncast`

cptartur opened the pull request:
<!-- Reference any GitHub issues resolved by this PR -->

Closes #

## Introduced changes

<!-- A brief description of the changes -->

- Voyager provider seems to be no longer available. Disabled it so we always use Blast

## Checklist

<!-- Make sure all of these are complete -->

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

cptartur committed:
Disable voyager provider in `sncast`

franciszekjob commented on crates/sncast/src/helpers/rpc.rs:
```diff
@@ -57,27 +55,13 @@ impl RpcArgs {
-        if seed % 2 == 0 {
-            return Self::Blast;
-        }
-        Self::Voyager
+        FreeProvider::Blast
     }
```
It's called `semi_random` and it always returns blast - I know that this PR is a hotfix but this should be changed at some point, either by adding other provider or refactoring the code so the method name is not confusing. Could you please link an issue here?

cptartur replied:
I'd want to introduce an alternative provider in the future, there exist some other. But don't have time to research them right now.

cptartur replied:
And generally rework this logic, so provider can be overriden by the user. Will do that post hotfix

franciszekjob replied:
Yes sure, let's just not forget about it 👍

cptartur committed:
Pin rust version to 1.86.0 due to `cairo_vm` incompatibility (#3352)

<!-- Reference any GitHub issues resolved by this PR -->

Closes #

## Introduced changes

<!-- A brief description of the changes -->

Interface of `is_multiple_of` changed in rust 1.87.0. This is a know
regression that won't be resolved, see
https://github.com/rust-lang/rust/issues/140020

Due to that, `cairo_vm` no longer compiles on later rust version. Until
they upgrade and we upgrade our version of `cairo_vm`, then we have to
stay at rust 1.86.0

Issue to remove the pinned version
https://github.com/foundry-rs/starknet-foundry/issues/3351

## Checklist

<!-- Make sure all of these are complete -->

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`
